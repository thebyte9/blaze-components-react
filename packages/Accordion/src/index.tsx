import React, { useEffect, useState } from "react";
import AccordionContent from "./AccordionContent";
import AccordionContentDetails from "./AccordionContent/AccordionContentDetails";
import AccordionContentFooter from "./AccordionContent/AccordionContentFooter";
import AccordionHeader from "./AccordionHeader";

interface IAccordionProps {
  children: [JSX.Element, JSX.Element];
  open?: boolean;
}

const Accordion = ({ children, open }: IAccordionProps): JSX.Element => {
  const flex = "flex";
  const none = "none";

  const [accordionStatus, setAccordionStatus] = useState<string>(
    open ? flex : none
  );

  const [header, content]: [JSX.Element, JSX.Element] = children;

  useEffect(() => {
    setAccordionStatus(open ? flex : none);
  }, [open]);

  const isActive: boolean = accordionStatus === flex;
  const toggleAccordion = (): void =>
    setAccordionStatus(isActive ? none : flex);

  return (
    <div className="accordion">
      <div className="accordion__header">
        {header}
        <div
          className="icon-button icon-button--accordion"
          onClick={toggleAccordion}
        >
          <i className="material-icons">keyboard_arrow_down</i>
        </div>
      </div>
      <div
        className="accordion__content-wrapper"
        style={{ display: accordionStatus }}
      >
        {content}
      </div>
    </div>
  );
};

Accordion.Content = AccordionContent;
Accordion.ContentDetails = AccordionContentDetails;
Accordion.ContentFooter = AccordionContentFooter;
Accordion.Header = AccordionHeader;
export default Accordion;
