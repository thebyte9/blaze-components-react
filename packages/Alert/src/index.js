import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@blaze-react/button';

const Alert = ({ children, close, icon, type, ...attrs }) => {
  const [offModal, setModalOff] = useState(false);
  const assignType = type && `alert--${type}`;
  const isDismissable = close && 'alert--dismissable';
  const withIcon = icon && 'alert--icon';

  const renderAlert = (
    <div className={`alert ${assignType} ${isDismissable} ${withIcon}`} {...attrs}>
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <Button onClick={() => setModalOff(true)} className="icon-button icon-button--close">
          <i className="material-icons">close</i>
        </Button>
      )}
    </div>
  );

  return <Fragment>{!offModal && renderAlert}</Fragment>;
};

Alert.propTypes = {
  icon: PropTypes.string,
  type: PropTypes.string,
  close: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Alert.defaultProps = {
  icon: '',
  type: '',
  close: false,
  children: 'No content'
};

export default Alert;
