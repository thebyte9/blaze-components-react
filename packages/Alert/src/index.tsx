import Button from "@blaze-react/button";
import React, { Fragment, FunctionComponent, useState } from "react";
interface IAlertProps {
  icon?: string;
  type?: string;
  close?: boolean;
  children?: JSX.Element | string | {} | [];
}

const Alert: FunctionComponent<IAlertProps> = ({
  children,
  close,
  icon,
  type,
  ...attrs
}): JSX.Element => {
  const [offModal, setModalOff] = useState(false);
  const assignType = type && `alert--${type}`;
  const isDismissable = close && "alert--dismissable";
  const withIcon = icon && "alert--icon";
  const renderAlert = (
    <div
      className={`alert ${assignType} ${isDismissable} ${withIcon}`}
      {...attrs}
    >
      {icon && <i className="material-icons">{icon}</i>}
      {children}
      {close && (
        <Button
          onClick={() => setModalOff(true)}
          className="icon-button icon-button--close"
        >
          <i className="material-icons">close</i>
        </Button>
      )}
    </div>
  );
  return <Fragment>{!offModal && renderAlert}</Fragment>;
};
Alert.defaultProps = {
  children: "No content",
  close: false,
  icon: "",
  type: ""
};
export default Alert;
