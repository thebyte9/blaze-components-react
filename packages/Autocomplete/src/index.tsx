import Input from "@blaze-react/input";
import utils from '@blaze-react/utils';
import React, { Fragment, useState } from "react";

interface IAutocompleteProps {
  data: {
    keyValue: string;
    filterBy: string[];
    data: object[];
  };
  utils: {
    uniqueId: (element: any) => string
  };
  selected: (...args: any[]) => any;
  placeholder?: string;
}
const Autocomplete: React.SFC<IAutocompleteProps> = ({
  data: { data, filterBy: keys, keyValue },
  placeholder,
  selected,
  utils: {
    uniqueId
  }
}) => {
  const [inputValue, setInputValue] = useState<string>('');
  const [showSelect, setShowSelect] = useState<boolean>(false);

  const handleChange = ({ value }: { value: string }): void => {
    setShowSelect(true);
    setInputValue(value);
    filterByValue(value);
  };

  const filterByValue = (value: string): object[] =>
    data.filter((copy: object) =>
      !!keys.some(key => copy[key].toLowerCase().includes(value.toLowerCase()))
    );

  const handleClick = (copiedData: object): void => {
    setInputValue(copiedData[keyValue]);
    setShowSelect(false);
    selected(copiedData);
  };

  return (
    <Fragment>

      <Input placeholder={placeholder} onChange={handleChange} value={inputValue} />

      {showSelect && filterByValue(inputValue).map((copiedData: object): JSX.Element => (
        <div
          className="panel"
          key={uniqueId(copiedData)}
          onClick={() => handleClick(copiedData)}
          role="button">{copiedData[keyValue]}
        </div>
      ))}

    </Fragment>
  );
};
Autocomplete.defaultProps = {
  placeholder: "Search",
  selected: (): void => {
    return;
  }
};
export default utils(Autocomplete);
