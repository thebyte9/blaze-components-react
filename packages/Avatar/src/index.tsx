import React, { FunctionComponent, useState } from "react";
interface IAvatarProps {
  modifier?: string;
  url?: string;
  username?: string;
  attrs?: any;
}
const Avatar: FunctionComponent<IAvatarProps> = ({
  modifier,
  url,
  username,
  ...attr
}): JSX.Element => {
  const [avatarUrl, setAvatar] = useState<string | undefined>(url);
  const [invalidUrl, setInvalidUrl] = useState<boolean>(true);
  const buildedModifier = modifier ? `avatar--${modifier}` : "";


  const getInitilas = (): string => {
    if (!username) {
      return '!';
    }
    return username
      .split(" ")
      .map(subName => subName[0].toUpperCase())
      .join("")
      .substring(0, 2);
  }

  const initials: string = getInitilas();

  if (url && invalidUrl) {
    const imageData = new Image();
    imageData.src = url;

    imageData.addEventListener('load', (): void => {
      setAvatar(url)
      setInvalidUrl(false);
    });
  }

  return (
    <div className={`avatar ${buildedModifier}`}>
      {avatarUrl ? (
        <img src={avatarUrl} alt="avatar" {...attr} />
      ) : (
          <span>{initials}</span>
        )}
    </div>
  );
};

Avatar.defaultProps = {
  modifier: "",
  url: "",
  username: ""
};
export default Avatar;
