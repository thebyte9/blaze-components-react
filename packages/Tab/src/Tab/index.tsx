import Button from "../../../Button/src";
import PropTypes from "prop-types";
import React, { useState } from "react";

type selectedType = string | number;

interface ITabProps {
  selected: selectedType;
  children?: any;
}

export const Tab = ({ selected, children }: ITabProps) => {
  const [selectedValue, setSelected] = useState(selected);

  return (
    <div className="tabs">
      <div className="tabs__list">
        {children.map(
          ({ props: { title = "Unnamed tab" } }: any, step: any) => (
            <Button
              className={`tabs__list-item ${
                step === selectedValue ? "current" : ""
              }`}
              onClick={() => setSelected(step)}
              key={title}
            >
              {title}
            </Button>
          )
        )}
      </div>
      {children[selectedValue]}
    </div>
  );
};

Tab.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  selected: PropTypes.number
};

Tab.defaultProps = {
  children: "No content",
  selected: 0
};
