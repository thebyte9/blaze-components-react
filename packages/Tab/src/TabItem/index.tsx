import PropTypes from "prop-types";
import React from "react";

interface ITabItemProps {
  action: () => Record<string, unknown>;
  children?: any;
  title?: string;
}

export const TabItem = ({ action, children }: ITabItemProps) => {
  action();
  return <div className="tabs__content current">{children}</div>;
};

TabItem.propTypes = {
  action: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

TabItem.defaultProps = {
  action: (): void => {
    return;
  },
  children: "No content"
};
