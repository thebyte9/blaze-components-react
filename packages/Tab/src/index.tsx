import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@blaze-react/button';

interface TabItemProps {
  action: () => {},
  children?: any
}

export const TabItem = ({ action, children }: TabItemProps) => {
  action();
  return <div className="tabs__content current">{children}</div>;
};

TabItem.propTypes = {
  action: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

TabItem.defaultProps = {
  action: () => { },
  children: 'No content'
};

type selectedType = string | number;

interface TabProps {
  selected: selectedType,
  children?: any
}

export const Tab = ({ selected, children }: TabProps) => {
  const [_selected, setSelected] = useState(selected);

  return (
    <div className="tabs">
      <div className="tabs__list">
        {children.map(({ props: { title = 'Unnamed tab' } }: any, step: any) => (
          <Button
            className={`tabs__list-item ${step === _selected ? 'current' : ''}`}
            onClick={() => setSelected(step)}
            key={title}>
            {title}
          </Button>
        ))}
      </div>
      {children[_selected]}
    </div>
  );
};

Tab.propTypes = {
  selected: PropTypes.number,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Tab.defaultProps = {
  selected: 0,
  children: 'No content'
};
