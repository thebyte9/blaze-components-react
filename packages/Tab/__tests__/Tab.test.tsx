import '@testing-library/jest-dom';

import { Tab, TabItem } from '../src';
import { fireEvent, render, screen } from '@testing-library/react';

import React from 'react';

const TabComponent = (
  <Tab selected={1}>
    <TabItem>Basic content here</TabItem>
    <TabItem title="Advanced">Advanced content here</TabItem>
    <TabItem title="Other">Other content here</TabItem>
  </Tab>
);

describe('Tab component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const { asFragment } = render(TabComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should navigate between tabs', () => {
    render(TabComponent);

    const button = screen.getByText('Advanced');
    fireEvent.click(button);

    screen.getByText('Advanced');
  });
});
