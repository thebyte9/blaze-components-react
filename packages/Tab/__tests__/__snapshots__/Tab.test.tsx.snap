// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component should be defined and renders correctly (snapshot) 1`] = `
<div
  className="tabs"
>
  <div
    className="tabs__list"
  >
    <Button
      className="tabs__list-item "
      disabled={false}
      key="Unnamed tab"
      modifiers=""
      onClick={[Function]}
      submit={false}
    >
      Unnamed tab
    </Button>
    <Button
      className="tabs__list-item current"
      disabled={false}
      key="Advanced"
      modifiers=""
      onClick={[Function]}
      submit={false}
    >
      Advanced
    </Button>
    <Button
      className="tabs__list-item "
      disabled={false}
      key="Other"
      modifiers=""
      onClick={[Function]}
      submit={false}
    >
      Other
    </Button>
  </div>
  <Component
    action={[Function]}
    title="Advanced"
  >
    Advanced content here
  </Component>
</div>
`;
